### DISCUSSION TITLE
Role / Permission Impersonation

### DISCUSSION THREAD

**Discussion by aglipanci** [spatie/laravel-permission#2613] [Q&A]
Hi there!
Has anyone tried to do some kind of impersonation for roles and/or permissions using this package?

What I have in mind is to be able to modify the permissions/roles relationships in runtime so that a user can behalf as if they have some other role and permissions.

Ideas are more than welcome.

Cheers!

**Comment by drbyte** [comment]
While this is not exactly an answer to your question about "how to simulate a different user's permissions while still being 'me'",
what I usually use when making an app allow to "impersonate the actual user" is https://github.com/404labfr/laravel-impersonate ... it allows you to suspend your own login session and login as the other person, and then return to yourself again later.

**Reply by aglipanci** [reply]
That's what I ended up doing, however, in terms of logging and audits having a role impersonating would be great because the user will always be the same.
